### PR TITLE
Links with hardcoded context

### PR DESCRIPTION
**Pull request by maximiliankolb**
This PR changes a couple of anchors and xrefs:

* Use {context} attribute for xrefs > do not hardcode "content-management" or "provisioning" if the xref could also just use the `{context}` attribute
* Add {context} attribute to anchors > mostly for consistency reasons
* Splits a couple of files that contain other anchors/xrefs without the `{context}` attribute

It's probably easiest to review the individual commits.

Cherry-pick into:

* [x] Foreman 3.2
* [x] Foreman 3.1

### CODE REVIEW

**Comment by maximiliankolb** [comment]
Rebased to master, hoping the checks will be green this time 🤞

**Comment by maximiliankolb** [comment]
Rebased to master & looking for a review :)

**Comment by melcorr** [comment]
Assigning to myself but it will be early next week.

File: guides/common/modules/proc_salt-using-content-management.adoc [hunk 1]
```diff
@@ -4,7 +4,7 @@
 Provide managed hosts with the required Salt Minion client software using {Project}.
 
 .Procedure
-. {ContentManagementDocURL}Content-Creating_a_Custom_Product[Create a product] called `Salt`.
+. {ContentManagementDocURL}Creating_a_Custom_Product_content-management[Create a product] called `Salt`.
```
Comment (melcorr) [inline guides/common/modules/proc_salt-using-content-management.adoc:7]: This doesn't work for me.

**Review by melcorr — CHANGES_REQUESTED** [review]
I clicked every  link and reviewed commit by commit, as recommended. 
The only thing is that the broken link is still broken :-/

**Pull request merged**